bot/util.go

```diff
@@ -265,26 +275,60 @@ func NodeID() string {
 
 // RefreshStatus updates the provided sessions status according to the current status set
 func RefreshStatus(session *discordgo.Session) {
-	var streamingURL string
-	var status string
-	err1 := common.RedisPool.Do(radix.Cmd(&streamingURL, "GET", "status_streaming"))
-	err2 := common.RedisPool.Do(radix.Cmd(&status, "GET", "status_name"))
+	var streamingUrl string
+	var statusText string
+	var gameType string
+	var statusTypeStr string
+	var statusType discordgo.Status
+	err1 := common.RedisPool.Do(radix.Cmd(&streamingUrl, "GET", "status_streaming_url"))
+	err2 := common.RedisPool.Do(radix.Cmd(&statusText, "GET", "status_name"))
+	err3 := common.RedisPool.Do(radix.Cmd(&gameType, "GET", "status_game_type"))
+	err4 := common.RedisPool.Do(radix.Cmd(&statusTypeStr, "GET", "status_type"))
+
 	if err1 != nil {
-		logger.WithError(err1).Error("failed retrieiving bot streaming status")
+		logger.WithError(err1).Error("failed retrieving bot streaming status")
 	}
 	if err2 != nil {
-		logger.WithError(err2).Error("failed retrieiving bot status")
+		logger.WithError(err2).Error("failed retrieving bot status")
+	}
+	if err3 != nil {
+		logger.WithError(err3).Error("failed retrieving bot game type")
+	}
+	if err4 != nil {
+		logger.WithError(err4).Error("failed retrieving bot status type")
 	}
 
-	if streamingURL != "" {
-		session.UpdateStreamingStatus(0, status, streamingURL)
+	switch statusTypeStr {
+	case "online":
+		statusType = discordgo.StatusOnline
+	case "idle":
+		statusType = discordgo.StatusIdle
+	case "dnd":
+		statusType = discordgo.StatusDoNotDisturb
+	case "offline":
+		statusType = discordgo.StatusOffline
+	default:
+		statusType = discordgo.StatusOnline
+	}
```

Comment:
I feel... weird about this `statusTypeStr` versus `statusType`, however
I cannot quite pinpoint *why* I feel weird about this part.

It probably is fine:tm:, though.

Reply:
You probably feel weird about it because it means there are two variables that are basically the same except for one is a string and the other is an enum value. I'm not really sure what the best way to convert is, or even if it's necessary to convert, I did it here mostly for precision later when it gets passed around, but it certainly could do with just a single check and then pass the string around instead of the enum value 🤷‍♀